src/test/java/com/microsoft/sqlserver/jdbc/NTLMConnectionTest.java

```diff
@@ -0,0 +1,413 @@
+package com.microsoft.sqlserver.jdbc;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.LogManager;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Constants;
+import com.microsoft.sqlserver.testframework.PrepUtil;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+
+/**
+ * A class for testing basic NTLMv2 functionality.
+ */
+@RunWith(JUnitPlatform.class)
+@Tag(Constants.xAzureSQLDW)
+@Tag(Constants.xAzureSQLDB)
+@Tag(Constants.xAzureSQLMI)
+@Tag(Constants.NTLM)
+public class NTLMConnectionTest extends AbstractTest {
+
+    private static SQLServerDataSource dsNTLMLocal = null;
+    private static String serverFqdn;
+    private static String connectionString = connectionStringNTLM;
+
+    @BeforeAll
+    public static void setUp() throws Exception {
+        // reset logging to avoid servere logs due to negative testing
+        LogManager.getLogManager().reset();
+
+        dsNTLMLocal = new SQLServerDataSource();
+        AbstractTest.updateDataSource(connectionString, dsNTLMLocal);
+
+        // grant view server state permission - needed to verify NTLM
+        try (Connection con = PrepUtil

```

Comment:
At this point, you have already executed `connectionString = connectionStringNTLM`, line 43. To be able to grant a permission, the user must have GRANT option, which NTLM user does not necessarily have. Please make sure you execute this statement with `sa` user (original connection string).